engine/class_modules/sc_demon_hunter.cpp

```diff
@@ -2625,6 +2651,11 @@ struct immolation_aura_t : public demon_hunter_spell_t
           p()->spawn_soul_fragment( soul_fragment::LESSER, 1 );
           p()->proc.soul_fragment_from_fallout->occur();
         }
+        if ( p()->sets->has_set_bonus( DEMON_HUNTER_VENGEANCE, T28, B2 ) && rng().roll( (p()->set_bonuses.t28_vengeance_2pc->effectN( 1 ).base_value()) ) )
```

Comment:
Same as the note above, maybe more advantageous in this case because it already requires set_bonuses.t28_vengeance_2pc spell data to be ok() for this line to function.

Also, I believe the rng().roll() here is using the incorrect value, should probably be using .percent() rather than .base_value(). Alternative, you can just use set_bonuses.t28_vengeance_2pc->proc_chance() since it is on the spell data for the proc.

> Name             : Burning Hunger (id=364454) [Spell Family (107), Passive, Hidden] 
> Proc Chance      : 10%